Atomic/backends/_docker.py

```diff
@@ -247,14 +265,17 @@ def _ping(self):
     def delete_image(self, image, force=False):
         return self.d.remove_image(image, force=force)
 
-    def update(self, name, force=False):
+    def update(self, name, force=False, **kwargs):
+        debug = kwargs.get('debug', False)
+        # Only delete containers if a new image is actually pulled.
```

Comment:
The comment was misplaced a little. I have moved it down.  If you use update with --force, existing containers based on the old image are deleted.  So in the case that a new image is not pulled, the containers should not be deleted.